src/main/java/org/subethamail/smtp/internal/command/BdatCommand.java

```diff
@@ -36,7 +39,7 @@ public void execute(String commandString, Session sess)
             return;
         }
 
-        InputStream stream = new BdatInputStream(sess.getRawInput(), sess, bdat.size, bdat.isLast);
+        InputStream stream = new BdatInputStream(new BufferedInputStream(sess.getRawInput(), BUFFER_SIZE), sess, bdat.size, bdat.isLast);
```

Comment:
Sorry, I wasn't clear enough. Please remove all stream stuff from this PR. Put this sort of change in another PR. In the back of my mind there was some reason that this didn't happen earlier, can't remember but I'll look at it closer in next PR.